homeassistant/components/light/hyperion.py

```diff
@@ -90,34 +140,107 @@ def turn_on(self, **kwargs):
         """Turn the lights on."""
         if ATTR_RGB_COLOR in kwargs:
             self._rgb_color = kwargs[ATTR_RGB_COLOR]
-        else:
+            self._rgb_mem = kwargs[ATTR_RGB_COLOR]
+        elif self._rgb_mem == [0, 0, 0]:
             self._rgb_color = self._default_color
+            self._rgb_mem = self._default_color
+        else:
+            self._rgb_color = self._rgb_mem
+
+        if ATTR_BRIGHTNESS in kwargs:
+            self._brightness = kwargs[ATTR_BRIGHTNESS]
 
+        if ATTR_EFFECT in kwargs:
+            self._effect = kwargs[ATTR_EFFECT]
```

Comment:
Same as above.

Reply:
The Effect case is special, Hyperion takes to long to activate an effect and update its internal status. Therefore I experianced that when you activate an effect, Hyperion will start activating that effect, but imediatly HASS will ask for an update of the status, and because Hyperion is not yet done with activating the effect it will return a json staing that Hyperion is OFF. in the second time an update is requested by HASS the correct json response will be sent by Hyperion and the state will display correctly.

Therfore HASS behaved as followed: you turn on an effect, HASS will imediatly show the light is turned off, and a few seconds later HASS will show the light is turned on again and the correct effect is displayed.

This is super anying, therfore I have used "self._skip_check = True", this will cause the first update HASS requests imidiatly after you select an effect to be aborted. The second and following update requests will be done as usual. If some strange thing happed, HASS will always correct the state in the second update request, but I have not seen that this was necassary.

To display the correct Effect in HASS imediatly after you select an effect I do need to update the self._ properties in the special case of effects.